src/sentry/issues/search.py

```diff
@@ -136,13 +150,37 @@ def _query_params_for_perf(
             conditions,
         )
 
-        mod_agg = list(aggregations).copy() if aggregations else []
-        mod_agg.insert(0, ["arrayJoin", ["group_ids"], "group_id"])
+        if group_ids:
+            transaction_conditions = [
+                [["hasAny", ["group_ids", ["array", group_ids]]], "=", 1],
+                *transaction_conditions,
+            ]
+            selected_columns = [
+                [
+                    "arrayJoin",
+                    [
+                        [
+                            "arrayIntersect",
+                            [
+                                ["array", group_ids],
+                                "group_ids",
+                            ],
```

Comment:
Since `group_ids` column is an array and could contain IDs not included in the postgres group_id list, we need to remove the group_ids before arrayJoining to filter out the extra group_ids from the result set.

Reply:
Have we tested this in production? One thought here is that we might have a really large list of group ids (up to 1000 I think?). Since these are arrays would it be scanning the whole array for every row?

Reply:
I included the `AND equals(hasAny(_snuba_group_ids, [1, 2]), 1)` `WHERE` predicate to filter down the rows not relevant to the group_ids. We'd still be doing the `arrayIntersect` though but I'd assume that operation would only be applied to the rows that at least satisfy what we're searching for.